packages/plugin-loans-api/src/models/utils/transactionUtils.ts

```diff
@@ -0,0 +1,573 @@
+    .sort({ payDate: 1 })
+    .lean<IScheduleDocument & any>();
+
+  return { preSchedule, nextSchedule };
+};
+
+export const getCalcedAmounts = async (
+  models: IModels,
+  subdomain: string,
+  doc: ICalcDivideParams
+) => {
+  let result: {
+    undue: number;
+    interestEve: number;
+    interestNonce: number;
+    total: number;
+    insurance: number;
+    debt: number;
+    payment: number;
+    preSchedule: any;
+  };
+  result = {
+    undue: 0,
+    interestEve: 0,
+    interestNonce: 0,
+    insurance: 0,
+    debt: 0,
+    payment: 0,
+    total: 0,
+    preSchedule: undefined
+  };
+
+  if (!doc.contractId) {
+    return result;
+  }
+
+  const trDate = getFullDate(doc.payDate);
+
+  const contract = await models.Contracts.getContract({
+    _id: doc.contractId
+  });
+
+  let { preSchedule, nextSchedule } = await getAOESchedules(models, contract);
+
+  if (!preSchedule && !nextSchedule) {
+    return result;
+  }
+
+  const startDate = getFullDate(contract.startDate);
+
+  if (trDate < startDate) {
+    return result;
+  }
+
+  if (!preSchedule) {
+    preSchedule = {
+      balance: contract.leaseAmount,
+      payDate: startDate,
+      insurance: nextSchedule.insurance,
+      debt: nextSchedule.debt
+    };
+  }
+
+  const prePayDate = getFullDate(preSchedule.payDate);
+  result.preSchedule = preSchedule;
+
+  // closed contract
+  if (!nextSchedule) {
+    const unduePercent =
+      (await getUnduePercent(models, subdomain, preSchedule.payDate)) ||
+      contract.unduePercent ||
+      0.2;
+    result.undue = Math.round(
+      ((preSchedule.balance * contract.interestRate) / 100 / 365) *
+        unduePercent *
+        getDiffDay(prePayDate, trDate)
+    );
+    const { diffEve, diffNonce } = getDatesDiffMonth(prePayDate, trDate);
+    result.interestEve = calcInterest({
+      balance: preSchedule.balance,
+      interestRate: contract.interestRate,
+      dayOfMonth: diffEve
+    });
+    result.interestNonce = calcInterest({
+      balance: preSchedule.balance,
+      interestRate: contract.interestRate,
+      dayOfMonth: diffNonce
+    });
+    result.insurance = preSchedule.insurance;
+    result.payment = preSchedule.balance;
+    return result;
+  }
+
+  // correct run
+  if (trDate < prePayDate) {
+    return result;
+  }
+
+  // one day two pay
+  if (getDiffDay(trDate, prePayDate) === 0) {
+    if (preSchedule.status === SCHEDULE_STATUS.LESS) {
+      result.undue = (preSchedule.undue || 0) - (preSchedule.didUndue || 0);
+      result.interestEve =
+        (preSchedule.interestEve || 0) - (preSchedule.didInterestEve || 0);
+      result.interestNonce =
+        (preSchedule.interestNonce || 0) - (preSchedule.didInterestNonce || 0);
+      result.insurance =
+        (preSchedule.insurance || 0) - (preSchedule.didInsurance || 0);
+      result.payment =
+        (preSchedule.payment || 0) - (preSchedule.didPayment || 0);
+      result.debt = (preSchedule.debt || 0) - (preSchedule.didDebt || 0);
+    }
+
```

Comment:
from started this row to 193th row in this section so strange this will be checked next time i think so this section is wrong